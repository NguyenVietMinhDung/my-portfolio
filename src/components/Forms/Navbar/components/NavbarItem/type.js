export type Props = {
  index: number,
  activatedIndex: number,
  name: string,
  url: string,
  activateNavigationItem: Function,
};
