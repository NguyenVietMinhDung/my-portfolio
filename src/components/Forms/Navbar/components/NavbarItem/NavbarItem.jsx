// @flow
import React, { memo } from 'react';
import CSSModules from 'react-css-modules';
import classNames from 'classnames';
import zenscroll from 'zenscroll';
import styles from './NavbarItem.scss';
import type { Props } from './type';

const getStyleClass = (
  index,
  activatedIndex,
) => (index === activatedIndex ? styles['is-active'] : '');

const activate = (
  activateNavigationItem,
  index,
  openNavigationMenu,
  itemId,
) => () => {
  const element = document.getElementById(itemId);
  if (element) {
    zenscroll.to(element);
  }
  activateNavigationItem(index);
  openNavigationMenu();
};

const NavbarItem = (props: Props) => {
  const {
    index, itemId, name, url, activateNavigationItem, activatedIndex, openNavigationMenu,
  } = props;
  return (
    <li className={classNames(styles['nav-item'], getStyleClass(index, activatedIndex))}>
      <a
        styleName="content"
        href={url}
        onClick={activate(activateNavigationItem, index, openNavigationMenu, itemId)}
      >
        {name}
      </a>
    </li>
  );
};

export default memo<Props>(CSSModules(NavbarItem, styles));
