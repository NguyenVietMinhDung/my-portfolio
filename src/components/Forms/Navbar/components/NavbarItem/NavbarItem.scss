@import '../../../../../styles/abstracts/variables';
@import '../../../../../styles/abstracts/mixins';

.nav-item {
	@media (min-width: $layout-breakpoint) {
		position: relative;

		&:hover,
		&.is-active {
			a {
				color: $violet-color;

				&::before {
					height: 45px;
				}
			}
		}

		&:not(:first-child) {
			margin-left: 30px;
		}

		a::before {
			position: absolute;
			top: -50px;
			left: 50%;

			width: 3px;
			height: 0;

			content: '';
			transition: height 0.3s ease-out 0s;

			background-color: $violet-color;
		}
	}

	@media (max-width: $layout-breakpoint) {
		margin: 4px 0;

		&.is-active {
			a {
				color: $violet-color;
			}
		}
	}

	.content {
		@include font(inherit, uppercase, $light-grey-color, inherit, 500);
	}
}
