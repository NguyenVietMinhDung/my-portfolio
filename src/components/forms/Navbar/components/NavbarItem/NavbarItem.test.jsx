import 'jsdom-global/register';
import React from 'react';
import { shallow, mount } from 'enzyme';
import NavbarItem from './NavbarItem';

const defaultProps = {
  index: 2,
  activatedIndex: 0,
  name: 'Home',
  url: '#',
  activateNavigationItem: jest.fn(),
  openNavigationMenu: jest.fn(),
};

describe('<NavbarItem />', () => {
  test('renders correctly', () => {
    const component = shallow(<NavbarItem {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
  test('handle click event', () => {
    const component = mount(<NavbarItem {...defaultProps} />);
    component.find('a').simulate('click');
    component.setProps({ activatedIndex: 2 });
    expect(component.props().activateNavigationItem).toBeCalledWith(component.props().index);
    expect(component.props().openNavigationMenu).toBeCalled();
  });
});
