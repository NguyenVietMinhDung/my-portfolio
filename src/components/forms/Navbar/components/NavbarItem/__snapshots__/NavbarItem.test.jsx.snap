// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<NavbarItem /> renders correctly 1'] = `
<li
  className="nav-item"
>
  <a
    className="content"
    href="#"
    onClick={[Function]}
  >
    Home
  </a>
</li>
`;
